llvm/include/llvm/Object/ELFObjectFile.h

```diff
@@ -292,6 +293,9 @@ template <class ELFT> class ELFObjectFile : public ELFObjectFileBase {
   const Elf_Shdr *DotSymtabSec = nullptr; // Symbol table section.
   const Elf_Shdr *DotSymtabShndxSec = nullptr; // SHT_SYMTAB_SHNDX section.
 
+  // Hold CREL relocations for SectionRef::relocations().
+  mutable SmallVector<SmallVector<Elf_Crel, 0>, 0> Crels;
```

Comment:
Are there any thread-safety concerns here?

It looks like this can be modified when obtaining an iterator? 

Just thinking of a use case where multiple threads given a const reference to an ELFObjectFile iterate through different CREL sections in the same object.

There may be no expectation that this would work though.

Reply:
To the best of my knowledge, ELFObjectFile is not used by multi-threading. The `mutable` member is not nice, which is to comply with the interface requirement by `llvm-objdump -r/-R`.